components/net/http_loader.rs

```diff
@@ -1417,6 +1420,35 @@ fn http_network_or_cache_fetch(
         *done_chan = None;
     }
 
+    fn prompt_user_for_credentials(
+        context: &FetchContext,
+    ) -> std::option::Option<std::string::String> {
+        let local_embedder_proxy: EmbedderProxy;
+        {
+            local_embedder_proxy = context.state.embedder_proxy.lock().unwrap().clone();
+        }
+        let (name_sender, name_receiver) = ipc::channel().unwrap();
+        let name_prompt = PromptDefinition::Input("Name".to_string(), "".to_string(), name_sender);
+        let name_msg = (
+            None,
+            EmbedderMsg::Prompt(name_prompt, PromptOrigin::Trusted),
+        );
+        local_embedder_proxy.send(name_msg);
+        let name_str = name_receiver.recv().unwrap()?;
+
+        let (pass_sender, pass_receiver) = ipc::channel().unwrap();
+        let pass_prompt =
+            PromptDefinition::Input("Password".to_string(), "".to_string(), pass_sender);
+        let pass_msg = (
+            None,
+            EmbedderMsg::Prompt(pass_prompt, PromptOrigin::Trusted),
+        );
+        local_embedder_proxy.send(pass_msg);
+        let pass_str = pass_receiver.recv().unwrap()?;
+
+        serde::export::Some(base64::encode(&format!("{}:{}", name_str, pass_str)))
```

Comment:
This can be `Some(base64::encode(&format!("{}:{}", name_str, pass_str)))` instead.